src/components/Auth/SignUp/SignUp.react.js

```diff
@@ -386,7 +385,7 @@ export default class SignUp extends Component {
               </div>
               <div style={{ width: '304px', margin: '10px auto 10px' }}>
                 <Recaptcha
-                  sitekey={CAPTCHA_KEY}
+                  sitekey="6LcUf3EUAAAAAH8yy0iTTorsgMOasLiTkVje33hu"
```

Comment:
Never do this ever again

Reply:
@anshumanv I have found the reason why this is not working. I mean why it goes **undefined** all of a sudden.

Look at **config.js**
``` 
import $ from 'jquery';
import urls from './Utils/urls';

const url = `${urls.API_URL}/aaa/getApiKeys.json`;

/* global module */

$.ajax({
  url: url,
  dataType: 'json',
  crossDomain: true,
  timeout: 3000,
  async: true,
}).done(function(output) {
  module.exports.MAP_KEY = output.keys.MAP_KEY;
  module.exports.CAPTCHA_KEY = '6LcUf3EUAAAAAH8yy0iTTorsgMOasLiTkVje33hu';
});

```

So the reason why it works when you use the signup button to go to the signup page from the login page is because this **$.ajax** call gets enough time to respond and enter the **.done** part of the promise, which then successfully exports the captcha key value.

But when you use the url of signup directly or refresh the page, the **$.ajax** does not get time to fullfill the promise and enter the **done** section, thus not getting a change to export the CAPTCHA_KEY value. The easiest fix would be to shift the CAPTCHA export statement out of the **.done** statement.

Whoever wrote the config file must have simply copy pasted some old code coz you see the use of **module.exports** instead of **export**.

@samagragupta Don't forget to give me credit for this. :wink:
I appreciate your effort though.